### PR TITLE
refactor prd reader history

### PR DESCRIPTION
**Pull request by CyanAutomation**
## Summary
- extract PRD reader history helpers
- refactor sidebar state and navigation handlers
- cover history edge cases and document PRD reader

## Testing
- `npx prettier . --check`
- `npx eslint .`
- `npm run check:jsdoc`
- `npx vitest run`
- `npx playwright test`
- `npm run check:contrast`

------
https://chatgpt.com/codex/tasks/task_e_68bd4f42393083269705408cecc5fe89

### CODE REVIEW

File: src/helpers/prdReader/history.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * 2. Extract index from state and invoke callback.
+ * 3. Return an unbind function.
+ *
+ * @param {(i:number) => void} selectDoc
+ * @returns {() => void}
+ */
+export function bindHistory(selectDoc) {
+  const handler = (e) => {
+    const i = e.state && typeof e.state.index === "number" ? e.state.index : null;
+    if (i !== null) selectDoc(i, false);
```
[the remark this answers is not in the document]
Comment (CyanAutomation) [reply src/helpers/prdReader/history.js:47]: @codex fix comments
